### PR TITLE
Tail call control flow

### PR DESCRIPTION
**Pull request by frabert**

### CODE REVIEW

**Comment by frabert** [comment]
next: figure out why some registers are not optimized out in the tail calls

File: lib/Declarations.cpp [hunk 1]
```diff
@@ -239,6 +224,50 @@ FunctionDecl::DeclareInModule(std::string_view name,
   return func;
 }
 
+size_t BasicBlockContext::GetParamIndex(const ParameterDecl &decl) const {
```
Comment (2over12) [inline lib/Declarations.cpp:227]: Can this be an equality predicate on ParameterDecl? Ie. we create an operator== on ParameterDecl, then it's just std::find. Didnt we bump anvill to c++ 20, i think default equality operators on ParameterDecl should work right?

File: lib/Lifters/BasicBlockLifter.cpp [hunk 2]
```diff
@@ -198,9 +192,11 @@ bool BasicBlockLifter::DoInterProceduralControlFlow(
     }
     return !cc.stop;
   } else if (std::holds_alternative<anvill::Return>(override)) {
-    remill::AddTerminatingTailCall(block, intrinsics.function_return,
-                                   intrinsics);
```
[the remark this answers is not in the document]
Comment (frabert) [reply lib/Lifters/BasicBlockLifter.cpp:206]: Added them, but I may have missed some usage

**Review by 2over12 — APPROVED** [review]
LGTM

**Pull request merged**